src/ORM/Connect/MySQLStatement.php

```diff
@@ -116,6 +116,11 @@ public function getIterator(): Traversable
             }
             yield $row;
         }
+        // Check for the method first since $this->statement is a mixed type
+        if (method_exists($this->statement, 'data_seek')) {
+            // Reset so the query can be iterated over again
+            $this->statement->data_seek(0);
+        }
```

Comment:
I'll extract this out into its own separate PR - this should have been part of 5.0.0 and reflects the same logic in `MySQLQuery::getIterator()` which allows the same query result (in this case the same result from `SQLSelect`) to be looped over multiple times.

Reply:
See https://github.com/silverstripe/silverstripe-framework/issues/10856 and its associated PR.